app/Jobs/SendSubscribedNotificationsJob.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace App\Jobs;
+
+use App\Mail\CheckInNotificationMail;
+use App\Models\PhoneNumber;
+use App\Models\Subscription;
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldBeUnique;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Queue\SerializesModels;
+use Illuminate\Support\Facades\Cache;
+use Illuminate\Support\Facades\Log;
+use Illuminate\Support\Facades\Mail;
+
+class SendSubscribedNotificationsJob implements ShouldQueue, ShouldBeUnique
+{
+    use Dispatchable, InteractsWithQueue, Queueable, SerializesModels;
+
+    public function __construct(
+        public PhoneNumber $phone_number,
+    )
+    {
+    }
+
+    public function uniqueId(): string
+    {
+        return $this->phone_number->id;
+    }
+
+    public function handle(): void
+    {
+        Cache::lock('send-notifications:'.$this->phone_number->id, 30)->get(function () {
+            $check_ins = $this->phone_number->not_notified_check_ins()->get();
+
+            if ($check_ins->count()) {
+                $this->phone_number->subscriptions()
```

Comment:
Should we fire a `SubscriptionWasNotified` event and do the rest of the code from this job in the event?

The Check In models themselves are created from events, so if we replay events in the future the notification_sent_at date is going to be null